README.md

```diff
@@ -33,16 +33,17 @@ CC Global Footer requires a `donation-url` attribute which is the URL used for t
 
 ### CC Global Header
 
-The CC Global Header has two required attributes, `base-url` and `donation-url` , which are the URLs used for the API call and the Donation button respectively. For a development environment, the `base-url` could be `http://127.0.0.1:8000`.
+The CC Global Header has two required attributes, `base-url` and `donation-url` , which are the URLs used for the API call and the Donation button respectively. There is one additional attribute `isDevelopmentMode` you can set which renders placeholder Menu Items if you are in a development environment. However, if you have WordPress setup correctly and you want to test this component with real Nav Menu items, you should not pass the `isDevelopmentMode` attribute. For a development environment, an example of a `base-url` is `http://127.0.0.1:8000`.
 
 ```html
 <cc-global-header
   base-url="http://127.0.0.1:8000"
   donation-url="http:/example.com"
+  is-development-mode
 />
 ```
 
-**Note:** be sure to enable WordPress permalinks in order for the CC Global Header to be able to retrieve the WordPress global menu via an API request.
+**Note:** be sure to enable WordPress permalinks in order for the CC Global Header to be able to retrieve the WordPress global menu via an API request. If for some reason the API call is unsuccesful, maybe due to Wordpress being unavaible, there are dummy Menu items that will display but this feature is available only for Development environments.
```

Comment:
we could display a little error message with a button asking the user to reload the page? I'm not sure what the best method will be. a modal will be overkill for something like this.

Reply:
Rather than reloading the page, the button could trigger another HTTP request since the menu would update reactively when the menu items arrive. So, we would consider refactoring the HTTP request logic to a method, r whatever equivalent exists in Vue, that updates the data.

Reply:
I have a concern about the UX. It may feel like too much work for the user to click a button just to have menu items display. I do admit that failing API calls will probably not occur a lot. So i am suggesting, how about we use a method as you have said, and then have this method be called in the background if the API call is failing, say each time it fails, we wait for 5 seconds and try again...since it's reactive it would update seamlessly and we won't have redirect the users focus to manually triggering it via a button?

Reply:
Sounds good. Perhaps we can show an alternative message to the user while the menu items load. E.g., "Loading menu items..."

Reply:
@TimidRobot suggests that 
- the loading or failure text should resemble the menu text. 
- we should avoid DDoSing ourselves by limiting the retries or using an [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff). (specifically, a truncated exponential backoff)

Reply:
Okay. But to be clear, by Loading or Failure text we are referring to what is rendered as the menu items right? 

Say when it's successful we have :
- `What we do`
- `Who we are` etc 
When it fails the placeholder text should just render these same menus right? But hardcoded instead from the API? Is my understanding correct?


Having an exponential back off is a great idea. It was something I was thinking of, how we can rate-limit the retries.

Reply:
> But to be clear, by Loading or Failure text we are referring to what is rendered as the menu items right?

The request failure text should briefly describe the HTTP error using simple language. The text should use the same font style as the top-level menu items. We don't want to show a placeholder menu, even one with a realistic structure, for the HTTP failure since it may be confusing or erroneous.

A placeholder menu for development/demonstration purposes can be added via a boolean property on the component, such as `use-placeholder-items`. That way, we have an explicit way to display a placeholder menu.

Reply:
Okay I get it now. 
We have a Boolean `is-development-mode`. So we can just change it to `use-placeholder-items` if it would make it's usage easily understood.

Reply:
In the above example, I used the attribute `use-placeholder-items` since it is more descriptive about the intended outcome than `is-development-mode`. Any similarly descriptive attribute will suffice.

The main principles here are to be explicit about our intended component behavior and to provide meaningful information to the end-user/developer when the HTTP request fails.

Reply:
> Having an exponential back off is a great idea. It was something I was thinking of, how we can rate-limit the retries.

Let's keep an eye towards simplicity and minimalism when limiting retries. E.g., the minimum-viable solution might not need a retry feature. If we do retry, let's make sure the code is readable and small 🙂

Reply:
@brylie I have a simple implementation to display error messages:

![image](https://user-images.githubusercontent.com/40151808/146176247-1bf19c0e-727f-491c-8849-b838c3d53fec.png)

Reply:
As concerns retries, I haven't added it yet since it's not a priority as of now, but I wrote this just to see that we could easily implement it:

```js

function getMenu(requestUrl){
  return axios.get(requestUrl).then((response) => (vm.menus = response.data))
}

  var retryCounter = 0
     vm.getMenu(requestUrl)
     .catch((error) => {
       vm.errorMessage = error.message
        while(retryCounter < 3)
          {
            setTimeout(function() {
              vm.getMenu(requestUrl)
            }, 5000);
            retryCounter+=1
          }
     })            
```